File: semestr_2/lista1_lab/LiczbyPierwsze/LiczbyPierwsze.java
package LiczbyPierwsze;

public class LiczbyPierwsze {
    int[] pierwsze;
    int ilosc_pierwszych;

    public LiczbyPierwsze(int n)
    {
        ilosc_pierwszych = n - 1;
        boolean[] sito_eratostenesa = new boolean[n + 1];
        
        for(int i = 0; i <= n; i++)
        {
            sito_eratostenesa[i] = true;
        }

        for(int i = 2; i * i <= n; i++)
        {
            if(sito_eratostenesa[i] == true)
            {
                for(int j = i * i; j <= n; j += i)
                {
                    if(sito_eratostenesa[j] == true)
                        ilosc_pierwszych--;
                    sito_eratostenesa[j] = false;
                }
            }
        }
    
        pierwsze = new int[ilosc_pierwszych];
        int i = 2, j = 0;
        while(i <= n && j < n)
        {
            if(sito_eratostenesa[i] == true)
            {
                pierwsze[j] = i;
                j++;
            }
            i++;
        }
    }


    public int liczba(int m)
    {
        return pierwsze[m];
    }
}

File: semestr_2/lista1_lab/LiczbyPierwsze.cpp
#include "LiczbyPierwsze.h"

#include <iostream>
#include <vector>
#include <exception>

LiczbyPierwsze::LiczbyPierwsze(int n)
{
    std::cout << n << std::endl;
    bool sito_eratostenesa[n + 1];

    for(int i = 0; i <= n; i++)
    {
        sito_eratostenesa[i] = true;
    }

    for(int i = 2; i * i <= n; i++)
    {
        if(sito_eratostenesa[i] == true)
        {
            for(int j = i * i; j <= n; j += i)
            {
                sito_eratostenesa[j] = false;
            }
        }
    }

    for(int i = 2; i <= n; i++)
    {
        if(sito_eratostenesa[i] == true)
        {
            pierwsze.push_back(i);
        }
    }
}

LiczbyPierwsze::~LiczbyPierwsze()
{

}

int LiczbyPierwsze::liczba(int m)
{
    return pierwsze.at(m);
}